fmn/consumer/consumer.py

```diff
@@ -55,13 +53,17 @@ def __call__(self, message: message.Message):
 
     async def handle_or_rollback(self, message: message.Message):
         await self._ready
-        try:
-            await self.handle(message)
-        except Exception:
-            await self.db.rollback()
-            raise
-        else:
-            await self.db.commit()
+        # Get a database session
+        async with async_session_maker() as db:
+            self._rules_cache.db = db
+            # Process message
+            try:
+                await self.handle(message)
+            except Exception:
+                await db.rollback()
+                raise
+            else:
+                await db.commit()
 
     async def handle(self, message: message.Message):
```

Comment:
Hmm, I don’t like storing ephemeral objects in object members to pass them around, we should rather pass it into the method explicitly (if needs be). I can think of two alternative ways:

* It doesn’t look as if `Consumer.handle()` is used outside of the class, so how about we mark it as not public (e.g. rename to `_handle()`) and pass in the session object explicitly.
* We could merge both methods into `handle` and make whether or not the transaction should be rolled back on error a parameter.

Reply:
Yeah, I'll modify handle, it seems better indeed.